python/lsst/ts/ledprojector/ledprojector_csc.py

```diff
@@ -0,0 +1,256 @@
+        asyncio.TimeoutError
+            If it takes longer than self.config.led.connect_timeout
+        """
+        await self.disconnect_led()
+        if self.led_controller is None:
+            if self.config is None:
+                raise ValueError(
+                    "CSC tried to use LEDController with config equal to None"
+                )
+            self.led_controller = LEDController(
+                config=self.config,
+                log=self.log,
+                simulate=self.simulation_mode,
+            )
+        await asyncio.wait_for(
+            self.led_controller.connect(),
+            timeout=self.led_controller.COMMUNICATION_TIMEOUT,
+        )
+
+    # TODO should we be catching errors
+    async def disconnect_led(self) -> None:
+        """Disconnect to the LabJack & delete the LEDController object
+
+        Raises
+        ------
+        Exception
+            If disconnect failed
+        """
+        try:
+            if self.led_controller is None:
+                return
+            await self.led_controller.disconnect()
+            # Delete the led controller because the config may change.
+            self.led_controller = None
+        except Exception as e:
+            self.log.warning(f"Failed to disconnect led; continuing: {e!r}")
+
+    async def start(self) -> None:
+        await super().start()
+
+    async def close_tasks(self) -> None:
+        """Close the CSC gratefully.
+
+        Disconnects the labjack, deletes LEDController object
+        Then closes tasks.
+
+        """
+        await self.disconnect_led()
+        await super().close_tasks()
+
+    # TODO should I just try to connect first if not connected
+    async def do_switchAllOff(self, data: types.SimpleNamespace) -> None:
+        """Switch off all LEDs.
+
+        Parameters
+        ----------
+        data : salobj.BaseMsgType
+            Command data; ignored.
+
+        Raises
+        ----------
+        salobj.ExpectedError
+            If labjack isnt connected first
+        """
+        self.assert_enabled()
+        if self.led_controller is None:
+            raise salobj.ExpectedError("Labjack not connected")
+        await self.led_controller.switch_all_leds_off()
+
+    # TODO should I just try to connect first if not connected
+    async def do_switchAllOn(self, data: types.SimpleNamespace) -> None:
+        """Switch on all LEDs.
+
+        Parameters
+        ----------
+        data : salobj.BaseMsgType
+            Command data; ignored.
+
+        Raises
+        ----------
+        salobj.ExpectedError
+            If labjack isnt connected first
+        """
+        self.assert_enabled()
+        if self.led_controller is None:
+            raise salobj.ExpectedError("Labjack not connected")
+        await self.led_controller.switch_all_leds_on()
+
+    # TODO should I just try to connect first if not connected
+    async def switchLEDs(
```

Comment:
Please, use `snake_case` (e.g. `switch_leds`) for method names. The CSC commands should be the only ones that follow the `do_<commandName>` pattern.